src/gc/impl/manual/gc.d

```diff
@@ -198,19 +197,16 @@ class ManualGC : GC
 
     void addRoot(void* p) nothrow @nogc
     {
-
-        Root r = {p};
-        roots.insertBack(r);
+        roots.insertBack(Root(p));
     }
 
     void removeRoot(void* p) nothrow @nogc
     {
-
-        for (size_t i = 0; i < roots.length; ++i)
+        foreach (ref r; roots)
```

Comment:
How does this work if `rt.util.container.Array` doesn't have an `opApply`?  Is it because it has `opSlice`? I would have used `foreach` otherwise.